util/perl/OpenSSL/config.pm

```diff
@@ -141,6 +141,7 @@ my $guess_patterns = [
     [ 'Rhapsody:.*',                'ppc-apple-rhapsody' ],
     [ 'Darwin:.*?:.*?:Power.*',     'ppc-apple-darwin' ],
     [ 'Darwin:.*?:.*?:x86_64',      'x86_64-apple-darwin' ],
+    [ 'Darwin:.*?:.*?:arm64',       'arm64-apple-darwin' ],
     [ 'Darwin:.*',                  'i686-apple-darwin' ],
```

Comment:
Change `i686` to `${MACHINE}`.  It's really neither here not there to default to "i686" these days, and especially when we have `${MACHINE}` to lean back on.  Even better, I'd say that the four Darwin lines can be collapsed to this:

``` perl
    [ 'Darwin:.*?:.*?:Power.*',     'ppc-apple-darwin' ],
    [ 'Darwin:.*',                  '${MACHINE}-apple-darwin' ],
```

Reply:
I believe 32-Bit Intel versions of MacOS X returned `i386` for `uname -m`, but were using at least Intel x86 Core Duo CPUs. So an extra line for that would make sense, similar to ppc.

Reply:
That's not really necessary.  See, there are two tables for figuring out what OpenSSL config target is the one to go for.  The first table (the one we look at here) basically converts the uname findings to the "standard" (GNUish) system triplet.  The second table converts from GNUish system triplets to OpenSSL's config target.  Here's the regexp in the second table:

``` perl
      [ 'i.86-apple-darwin.*',
```

So as you can see, it doesn't really matter if the GNUish system triplet say `i386-apple-darwin` or `i686-apple-darwin`, because they're both treated the same in the end anyway...  and if we want to be correct from a GNUish system triplet, it's absolutely correct that 'Darwin:.*?:.*?:i386' maps to `i386-apple-darwin` rather than `i686-apple-darwin`...  if we actually made a difference between i386 and i686, it would be patently wrong to present i386 as i686 at this level.